client/analysisCtrl.ts

```diff
@@ -1140,7 +1044,7 @@ export default class AnalysisController {
             }
 
             // Janggi pass and Sittuyin in place promotion on Ctrl+click
-            if (this.chessground.state.stats.ctrlKey && 
+            if (this.chessground.state.stats.ctrlKey &&

```

Comment:
ok - leaving it as is - too big pr already and outside the scope anyway.

Those 2 classes indeed have a lot of overlap. 

My gut feeling tells me maybe best to still keep 2 separate classes but instead extract common code in a 3rd place to be reused and have those 2 classes be much shorter, instead of unifying them into 1 single classes with "if"-s depending on some "mode", but haven't really thought about it - just sharing my immediate thoughts. 

I have a strong obsession with averting cyclomatic complexity at all costs even if it results in a bit more lines of code and some code repetition so could not help myself pitching my 2 cents. 

But don't even know what your plan is exactly so maybe my thoughts not relevant.